cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_controller.go

```diff
@@ -152,30 +152,17 @@ func (c *machineController) run(stopCh <-chan struct{}) error {
 
 // findMachineByNodeProviderID find associated machine using
 // node.Spec.ProviderID as the key. Returns nil if either the Node by
-// node.Spec.ProviderID cannot be found or if the node has no machine
-// annotation. A DeepCopy() of the object is returned on success.
+// node.Spec.ProviderID cannot be found or if there is no machine which has matching machine.Spec.ProviderID
 func (c *machineController) findMachineByNodeProviderID(node *apiv1.Node) (*v1beta1.Machine, error) {
-	objs, err := c.nodeInformer.GetIndexer().ByIndex(nodeProviderIDIndex, node.Spec.ProviderID)
+	machines, err := c.machineInformer.Lister().Machines("").List(labels.Everything())
```

Comment:
Let's add an index (for provider-id) on Machines - this will make this cheap again.